quinn/src/platform/unix.rs

```diff
@@ -27,26 +29,30 @@ type IpTosTy = libc::c_int;
 #[derive(Debug)]
 pub struct UdpSocket {
     io: AsyncFd<mio::net::UdpSocket>,
+    last_send_error: Instant,
 }
 
 impl UdpSocket {
     pub fn from_std(socket: std::net::UdpSocket) -> io::Result<UdpSocket> {
         socket.set_nonblocking(true)?;
         let io = mio::net::UdpSocket::from_std(socket);
         init(&io)?;
+        let now = Instant::now();
         Ok(UdpSocket {
             io: AsyncFd::new(io)?,
+            last_send_error: now.checked_sub(2 * IO_ERROR_LOG_INTERVAL).unwrap_or(now),
         })
     }
 
     pub fn poll_send(
-        &self,
+        &mut self,
```

Comment:
We should keep the type signature of the fallback module in sync with this.

Reply:
Oops, I initially read this as also implement the logic in the fallback module. Well, done both now.